pkg/store/consul.go

```diff
@@ -97,9 +97,14 @@ func (s *Consul) setEphemeralTTL(ttl time.Duration) error {
 
 // CreateEphemeralSession creates the a global session
 // once that is used to delete keys at node failure
-func (s *Consul) createEphemeralSession() error {
+func (s *Consul) createEphemeralSession(key string) error {
 	s.Lock()
 	defer s.Unlock()
+
+	// Delete the key first (erase any attached session for that key)
+	// FIXME recover gracefully with the existing global session
+	s.Delete(key)
```

Comment:
uhm, this will delete **every time**, not just the first time we create the ephemeral session - not sure if we could do it differently